src/main/java/de/hysky/skyblocker/skyblock/dungeon/CroesusProfit.java

```diff
@@ -0,0 +1,272 @@
+            if (stack != null && stack.getNbt() != null && stack.getName().toString().contains("Chest")) {
+                long value = valueChest(stack);
+                if (value > bestValue) {
+                    secondBestChest = bestChest;
+                    secondBestValue = bestValue;
+                    bestChest = stack;
+                    bestValue = value;
+                } else if (value > secondBestValue) {
+                    secondBestChest = stack;
+                    secondBestValue = value;
+                }
+            }
+        }
+
+        for (Map.Entry<Integer, ItemStack> entry : slots.entrySet()) {
+            ItemStack stack = entry.getValue();
+            if (stack != null && stack.getNbt() != null) {
+                if (stack.equals(bestChest)) {
+                    highlights.add(ColorHighlight.green(entry.getKey()));
+                } else if (stack.equals(secondBestChest) && secondBestValue > dungeonKeyPriceData) {
+                    highlights.add(ColorHighlight.yellow(entry.getKey()));
+                }
+            }
+        }
+        return highlights;
+    }
+
+
+    private long valueChest(@NotNull ItemStack chest) {
+        long chestValue = 0;
+        int chestPrice = 0;
+        List<String> chestItems = new ArrayList<>();
+
+        Matcher matcher = CHEST_PATTERN.matcher(chest.getName().toString());
+        String chestType = (matcher.find() ? matcher.group(1) : "Error");     // we are doing it for sole reason of saving us headache of a spirit pet rarity
+
+        boolean processingContents = false;
+        for (Text line : chest.getTooltip(null, TooltipContext.BASIC)) {
+            String lineString = line.getString();
+            if (lineString.contains("Contents")) {
+                processingContents = true;
+                continue;
+            } else if (lineString.isEmpty()) {
+                processingContents = false;
+            } else if (lineString.contains("Coins") && !processingContents) {
+                chestPrice = Integer.parseInt(lineString.replaceAll(",", "").replaceAll("\\D", ""));
+            }
+
+            if (processingContents) {
+                if (lineString.contains("Essence")) {
+                    matcher = ESSENCE_PATTERN.matcher(lineString);
+                    if (matcher.matches()) {    // add to chest value result of multiplying price of essence on it's amount
+                        chestValue += getItemPrice(("ESSENCE_" + matcher.group("type")).toUpperCase()) * Integer.parseInt(matcher.group("amount"));
+                    }
+                } else {
+                    if (lineString.contains("Spirit")) {    // I hate pets in chests. Speaking of which, this is only part of code I'm really not sure of
+                        if (Objects.equals(chestType, "Diamond")) {
+                            chestValue += getItemPrice("Spirit Epic");      // Diamond chest containts epic spirit
```

Comment:
Instead of depending on an implementation detail (chest type) replace all of this with a check for the text colour being dark purple which is far more reliable and should require less code.

Reply:
> I originally thought of it, but due to some dark magic involved all we have here is plain text, according to what my debug were telling me (as well as by fact that whole thing, well, works)
> 
> I did made it more readable tho

Did you try reading the `Style` from the Text object? That part of the code seems to have access to it.